components/director/internal/domain/api/converter_test.go

```diff
@@ -1 +1,222 @@
-package api
+package api_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/kyma-incubator/compass/components/director/internal/domain/api"
+	"github.com/kyma-incubator/compass/components/director/internal/domain/api/automock"
+
+	"github.com/kyma-incubator/compass/components/director/internal/graphql"
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConverter_ToGraphQL(t *testing.T) {
+	// given
+	testCases := []struct {
+		Name     string
+		Input    *model.APIDefinition
+		Expected *graphql.APIDefinition
+	}{
+		{
+			Name:     "All properties given",
+			Input:    fixDetailedModelApiDefinition(t, "foo", "Foo", "Lorem ipsum", "group"),
+			Expected: fixDetailedGQLApiDefinition(t, "foo", "Foo", "Lorem ipsum", "group"),
+		},
+		{
+			Name:  "Empty",
+			Input: &model.APIDefinition{},
+			Expected: &graphql.APIDefinition{
+				Spec: &graphql.APISpec{
+					FetchRequest: &graphql.FetchRequest{},
+				},
+				Version: &graphql.Version{},
+				Auth:    &graphql.RuntimeAuth{},
+				Auths:   []*graphql.RuntimeAuth{},
+			},
+		},
+		{
+			Name:     "Nil",
+			Input:    nil,
+			Expected: nil,
+		},
+	}
+
+	for i, testCase := range testCases {
+		t.Run(fmt.Sprintf("%d: %s", i, testCase.Name), func(t *testing.T) {
+			// when
+			authConverter := &automock.AuthConverter{}
+			frConverter := &automock.FetchRequestConverter{}
+
+			if testCase.Input != nil {
```

Comment:
tests should be extremely simple, so we should avoid if's (nested if's even more)